### DISCUSSION TITLE
New Feature Request: Zim Scrapbook

### DISCUSSION THREAD

**Discussion by jerger** [zim-desktop-wiki/zim-desktop-wiki#2049] [Ideas]
Hi,

as very happy zim user and unhappy Firefox-Scrapbook-User (Scrapbook is no longer supported on no ffox) I found the following: https://www.onenote.com/clipper

What do you think about a Zim-Scrapbook Plugin ?
I think this could be in close range?

Best Regards, jerger

**Comment by Osndok** [comment]
This closely fits one of my zim workflows, so I am certainly interested here.

However, since Zim does not actually accept HTML content, I'm not sure a plugin is 100% needed (i.e. it would only be a minor gain/acceleration over what is currently available).

In particular, you can already copy/paste images & text, I have a plugin for a one-click region screenshot, and the 'template' plugin lets you copy markdown-formatted text (complete with links and source url).

**Comment by jerger** [comment]
With manual process I tried out:
1.) fetch Webpage with scrapbook (html + js + css + img-resources - goes to .mozilla/profile-xy/ScrapBook/id/...)
2.) copy scraped files manually to a zim notebook page subdirectory
3.) add the zim pages (name.txt) for subdirectory containing webpage-files manually (name corresponds to a scrapbook boockmark entry)
4.) tried to add a link to index.html file & some tags into the zim page ...

So creating a structure (subdir / name.txt) and fetching the metadata (tags) we could automate. Either on ffox plugin side (maybe fork a zim-scrapbook? ) or on zim side (import scrapbook-content from filesystem).

My open questions at the moment are:
* what are the limits of zim page-names (as scrapbook bookmark names are less restrictive, I think) ?
* How do binary files integrate in a git repository backend?
* As ffox no longer allows to write files to filesystem, we will need some kind of "host application" service between the plugin and zim ... 

If you like this idea, the next step could be to contact the scrapbook-plugin maintainer ?

**Comment by sojusnik** [comment]
> In particular, you can already copy/paste images & text, I have a plugin for a one-click region screenshot, and the 'template' plugin lets you copy markdown-formatted text (complete with links and source url).

@Osndok  Can you elaborate more on the "template" plugin? Can't find it myself. Or do you mean the templates when exporting notes?

**Comment by StephGbzh** [comment]
You should take a look at [zim-clip](https://addons.mozilla.org/en-US/firefox/addon/zim-clip/). It works on Firefox but also on Chromium and probably Chrome.

**Comment by jaap-karssenberg** [comment]
This sounds like it still needs a good brainstorm. Please use the "discussions" forum
